lib/company.rb

```diff
@@ -0,0 +1,150 @@
+require 'sinatra'
+require_relative '../models/models'
+require 'sinatra/activerecord'
+require_relative '../tools/db_controller'
+
+set :database, { adapter: 'sqlite3', database: '../db/base/development.sqlite3' }
+
+use Rack::Auth::Basic, 'Restricted' do |username, password|
+  [username, password] == %w(superadmin 12345)
+end
+
+get '/locations' do
+  DBController.locations
+end
+
+get '/projects' do
+  DBController.projects
+end
+
+get '/employees' do
+  DBController.employees
+end
+
+get '/locations/:id' do
+  DBController.find_location_by({ id: params['id'] })
+end
+
+get '/locations/:name' do
+  DBController.find_location_by({ name: params['name'] })
+end
+
+get '/projects/:id' do
+  DBController.find_project_by({ id: params['id'] })
+end
+
+get '/projects/:name' do
+  DBController.find_project_by({ name: params['name'] })
+end
+
+get '/employees/:id' do
+  DBController.find_employee_by({ id: params['id'] })
+end
+
+get '/employees/:name' do
+  DBController.find_employee_by({ name: params['name'] })
+end
+
+get '/employees/:surname' do
+  DBController.find_employee_by({ surname: params['surname'] })
+end
+
+post '/locations' do
+  begin
+    DBController.create_location(params[:location])
+    status 201
+  rescue StandardError => e
+    puts "StandardError #{e.message}"
+  end
+end
+
+post '/employees' do
+  begin
+    DBController.create_employee(params[:employee])
+    status 201
+  rescue StandardError => e
+    puts "StandardError #{e.message}"
+  end
+end
+
+post '/projects' do
+  begin
+    DBController.create_project(params[:project])
+    status 201
+  rescue StandardError => e
+    puts "StandardError #{e.message}"
+  end
+end
+
+patch '/locations/:name' do
+  begin
+    DBController.edit_location({ name: params['name'] }, { name: params['name_new'] })
+    status 200
+  rescue StandardError => e
+    puts "StandardError #{e.message}"
+  end
+end
+
+patch '/projects/:name' do
+  begin
+    DBController.edit_project({ name: params['name'] }, { name: params['name_new'] })
+    status 200
+  rescue StandardError => e
+    puts "StandardError #{e.message}"
+  end
+end
+
+patch '/employees/:name' do
+  begin
+    DBController.edit_employee_by({ name: params['name'] }, { name: params['name_new'] })
```

Comment:
Will be better to have name of parameter 'new_name' instead of 'name_new'.
And for other new_surname, new_email, etc.